pennylane/gradients/finite_difference.py

```diff
@@ -154,6 +154,211 @@ def finite_diff_coeffs(n, approx_order, strategy):
+
+        return [], lambda _: tuple(list_zeros)
+
+    gradient_tapes = []
+    shapes = []
+    c0 = None
+
+    coeffs, shifts = finite_diff_coeffs(n=n, approx_order=approx_order, strategy=strategy)
+
+    if 0 in shifts:
+        # Finite difference formula includes a term with zero shift.
+
+        if f0 is None:
+            # Ensure that the unshifted tape is appended
+            # to the gradient tapes, if not already.
+            gradient_tapes.append(tape)
+
+        # Store the unshifted coefficient. We know that
+        # it will always be the first coefficient due to processing.
+        c0 = coeffs[0]
+        shifts = shifts[1:]
+        coeffs = coeffs[1:]
+
+    method_map = choose_grad_methods(diff_methods, argnum)
+
+    for i, _ in enumerate(tape.trainable_params):
+        if i not in method_map or method_map[i] == "0":
+            # parameter has zero gradient
+            shapes.append(0)
+            continue
+
+        g_tapes = generate_shifted_tapes(tape, i, shifts * h)
+        gradient_tapes.extend(g_tapes)
+        shapes.append(len(g_tapes))
+
+    def processing_fn(results):
+        grads = []
+        start = 1 if c0 is not None and f0 is None else 0
+        r0 = f0 or results[0]
+
+        output_dims = []
+        for m in tape.measurements:
+            if m.return_type is qml.measurements.Probability:
+                output_dims.append(2 ** len(m.wires))
+            else:
+                output_dims.append(1)
+
+        for s in shapes:
+
+            if s == 0:
+                # parameter has zero gradient
+                if not isinstance(results[0], tuple):
+                    g = qml.math.zeros_like(results[0])
+                else:
+                    g = []
+                    for i in output_dims:
+                        g.append(qml.math.zeros(i))
+                grads.append(g)
+                continue
+
+            res = results[start : start + s]
+            start = start + s
+            # compute the linear combination of results and coefficients
+
+            # First compute the multiplication with coeff
+            linear_comb = []
+            for i, c in enumerate(coeffs):
+                if isinstance(res[i], (tuple, list)):
+                    elem = [r * c for r in res[i]]
+                else:
+                    elem = [res[i] * c]
+                linear_comb.append(elem)
+
+            # Second add all the terms for each measurement separately
+            pre_grads = []
+
+            for i in range(len(tape.measurements)):
+                elem = sum(r[i] for r in linear_comb)
+                pre_grads.append(elem)
+
+            # Add on the unshifted term
+            if c0 is not None:
+                # unshifted term
+                if isinstance(r0, (tuple, list)):
+                    c0r0 = [c0 * r for r in r0]
+                else:
+                    c0r0 = [c0 * r0]
+                pre_grads = [i + j for i, j in zip(pre_grads, c0r0)]

```

Comment:
Similar to above. If `pre_grads` is computed with `qml.math.tensordot`, then there's a way to do this computation using `qml.math` functions instead of loops, right?